py/objarray.c

```diff
@@ -504,13 +505,18 @@ STATIC mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value
                         // TODO: alloc policy; at the moment we go conservative
                         o->items = m_renew(byte, o->items, (o->len + o->free) * item_sz, (o->len + len_adj) * item_sz);
                         o->free = len_adj;
-                        dest_items = o->items;
+                        if (o->items != dest_items) { // m_renew moved o->items
```

Comment:
I don't think this check is needed, the code in this `if` can just be executed unconditionally.

Reply:
ah yeah, ofc! New patch incoming.